src/librustc/middle/resolve_lifetime.rs

```diff
@@ -1573,7 +1573,7 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
             .collect();
 
         // ensure that we issue lints in a repeatable order
-        def_ids.sort_by_key(|&def_id| self.tcx.def_path_hash(def_id));
+        def_ids.sort_by_cached_key(|&def_id| self.tcx.def_path_hash(def_id));
```

Comment:
As you noted, I think this is already cached -- it's something we do a lot in incremental. You agree, @michaelwoerister ?

Reply:
It's cached, yes, but still not entirely trivial. I.e. it has to do the local/upstream crate dispatch and for upstream `DefIds` it has to go through a number of function calls: https://github.com/rust-lang/rust/blob/780658a464603fa755d94b27f72a375bd81d07ea/src/librustc/ty/context.rs#L1325

So, I can imagine this being a win for sorting where each key might be compared against multiple times.